### PR TITLE
verovio 3.12.1 (new formula)

### PR DESCRIPTION
**Pull request by ahankinson**
Attempt number 2 at adding the Verovio command-line notation engraver.

Initial attempt in PR #112475

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/verovio.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  desc "Command-line MEI music notation engraver"
+  homepage "https://www.verovio.org"
+  url "https://github.com/rism-digital/verovio/archive/refs/tags/version-3.12.1.tar.gz"
+  sha256 "7f69b39e25f9662185906c9da495437e09539a520b9dda80f1bef818e905881b"
+  license "LGPL-3.0-only"
+  head "https://github.com/rism-digital/verovio.git", branch: "develop"
+
+  depends_on "cmake" => :build
+
+  resource("testdata") do
```
Comment (SMillerDev) [inline Formula/verovio.rb:11]:
```suggestion
  resource "homebrew-testdata" do
```

File: Formula/verovio.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  def install
+    system "cmake", "-S", "./cmake", "-B", "tools", *std_cmake_args
+    system "cmake", "--build", "tools"
+    system "cmake", "--install", "tools"
+  end
+
+  test do
+    system bin/"verovio", "--version"
+    resource("testdata").stage do
```
Comment (SMillerDev) [inline Formula/verovio.rb:24]:
```suggestion
    resource("homebrew-testdata").stage do
```

**Comment by ahankinson** [comment]
The failing test file has been fixed so this is now ready for re-review.

File: Formula/verovio.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    system "cmake", "--build", "tools"
+    system "cmake", "--install", "tools"
+  end
+
+  test do
+    system bin/"verovio", "--version"
+    resource("homebrew-testdata").stage do
+      shell_output("#{bin}/verovio Ahle_Jesu_meines_Herzens_Freud.mei -o #{testpath}/output.svg")
+      assert_predicate testpath/"output.svg", :exist?
+    end
```
Comment (SMillerDev) [inline Formula/verovio.rb:27]: This is unrelated to the resource
```suggestion
    end
    assert_predicate testpath/"output.svg", :exist?
```

**Comment by ahankinson** [comment]
Fixed (Sorry, I couldn't find any great resources to help me with writing the tests, so thank you for helping!)

**Review by SMillerDev — APPROVED** [review]
Thanks @ahankinson  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Review by chenrui333 — APPROVED** [review]
